### PR TITLE
php-fpm for php-7.0

### PR DESCRIPTION
**Pull request by popstas**
issue phansible/phansible#250

### CODE REVIEW

**Comment by popstas** [comment]
Travis CI failed because I used reference to variable from role-php, how to better solve this, maybe simple add php.ppa to role-nginx defaults?

**Comment by naxhh** [comment]
About the ppa. If you think the client should be able to edit this. Default is ok.

If you want to "hide" this from the client other approach should be used.
I didn't decided what should we do yet.

BTW we didn't finished fixing some roles so yes. CI failing is normal. but thanks for fixing it :)

**Comment by popstas** [comment]
I think that client don't need for change php ppa manually. But how to hide it? 
At this moment, php.ppa always present in ansible/vars/all.yml, but isolated role-nginx tests fails.

**Comment by naxhh** [comment]
At the end what you want to do is to allow the user to specify which version of php is using. So you could determine where the /var/run should point to.

Maybe we should allow

php:
  version: 5/7

instead of ppa.
Or just allow a full path and default to php7 one.

but! as I want to deploy this with the current release. Maybe that's a change we can keep for next release and now just expose as it is..

What do you think @debo ?

File: templates/default.tpl [hunk 1]
```diff
@@ -19,7 +19,7 @@ server {
 
     location ~ \.php$ {
         fastcgi_split_path_info ^(.+\.php)(/.+)$;
-        fastcgi_pass unix:/var/run/php5-fpm.sock;
+        fastcgi_pass unix:{% if php.ppa == 'php-7.0' %}/var/run/php/php7.0-fpm.sock{% else %}/var/run/php5-fpm.sock{% endif %};
```
Comment (debo) [inline templates/default.tpl:22]: @popstas as I mentioned in the other PR, would it be possible to to use a dynamically set fact also in this case? I don''t like the fact that nginx has to access variables from a different role which might not exists.

**Comment by debo** [comment]
@popstas you can easily run a shell command to identify whether PHP is installed or not and which version is used so you can decide to conditionally include what needed.

**Comment by popstas** [comment]
@debo, you mean this?

``` yml
    - name: get php version
      shell: 'php --version | grep -o "^PHP [0-9]"'
      register: php_version_command

    - name: set php_version fact
      set_fact: php_version="{{php_version_command.stdout}}"

    - name: debug
      debug: var=php_version # 'PHP 5' or 'PHP 7'
```

In current phansible playbook.yml nginx role plays before php role and playbook will failed.
So, if we will follow this way, we will make role nginx more depedent from php role.

@naxhh, I think that `php.version: 5/7` is better that check `php.ppa` variable. But I assume that php-7.0 will change to php-7.1, in this case just 5/7 will not be enough.

**Comment by debo** [comment]
Yes I thought something like that. The fact is that the nginx role should be able to run whether php is present or not I suppose. So, maybe, we could use that plus `when` to make sure the php template is set only if php is actually present. As per the role orders maybe php should come first... I'm not sure. I have to think about it.

**Pull request closed without merging**